### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by ecoologic**

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -1,26 +1,28 @@
-module Admin
-  class UsersController < ::AdminController
-    include Users
+module Admin::UsersController < ::AdminController
```
Comment (buitranquyet) [inline app/controllers/admin/users_controller.rb]: I thought it should be `class Admin::UsersController `

File: app/views/layouts/admin.html.haml [hunk 2]
```diff
@@ -1,5 +1,4 @@
 !!!
-%html
```
Comment (buitranquyet) [inline app/views/layouts/admin.html.haml]: 👍 This break turbolinks, took me lots of time to figure out (in Edward which use this app as template). 
But forgot to remove that last time

File: app/controllers/admin/users_controller.rb (hunk 1 above)
Comment (ecoologic) [inline app/controllers/admin/users_controller.rb]: oh gosh, you're right 😊

File: app/views/layouts/admin.html.haml (hunk 2 above)
Comment (ecoologic) [inline app/views/layouts/admin.html.haml]: the capybara specs as well

when we come around these kind of easy wins, great if we remember to update the app (;

File: Gemfile [hunk 3]
```diff
@@ -34,15 +34,20 @@ gem 'sass-rails', '~> 5.0'
...
+  gem 'dotenv-rails'
+
+  # Console
   gem 'byebug'
+  gem 'pry-meta'
+  gem 'pry-doc'
+  gem 'pry-rails'
+  gem 'pry-stack_explorer'
+  gem 'spring'
+  gem 'spring-commands-rspec'
```
Comment (ecoologic) [inline Gemfile]: spring is awesome guys, remember it to shave a few secs every time you run a spec

File: app/views/shared/_footer.html.haml [hunk 4]
```diff
@@ -1,5 +0,0 @@
-%hr
-  %p.text-center
-    Created by
-    = link_to 'NetEngine', 'http://netengine.com.au'
-    2016
```
Comment (ecoologic) [inline app/views/shared/_footer.html.haml]: btw, for sharing partials, we should user `app/views/application`, to take advantage of template inheritance (:

File: .rubocop.yml [hunk 5]
```diff
@@ -26,7 +26,7 @@ Lint/HandleExceptions:
     - 'bin/*'
 
 Metrics/AbcSize:
-  Max: 15
+  Max: 25
```
Comment (lucascaton) [inline .rubocop.yml]: Agreed!

File: Gemfile [hunk 6]
```diff
@@ -19,10 +17,12 @@ gem 'bugsnag'
...
 
 # Front-End
-gem 'uglifier', '>= 1.3.0'
+gem 'uglifier'
 gem 'jquery-rails'
 gem 'haml-rails'
-gem 'sass-rails', '~> 5.0'
+gem 'sass-rails'
+
+gem 'rails_12factor' # Not only for Heroku
```
Comment (lucascaton) [inline Gemfile]: We should remove this gem: https://github.com/heroku/rails_12factor#rails-5-and-beyond

File: Gemfile [hunk 7]
```diff
@@ -34,15 +34,20 @@ gem 'sass-rails', '~> 5.0'
...
+  gem 'pry-meta'
+  gem 'pry-doc'
+  gem 'pry-rails'
+  gem 'pry-stack_explorer'
+  gem 'spring'
+  gem 'spring-commands-rspec'
 
-  # Code analyzers
+  # Code Analyzers
+  gem 'massa'
```
Comment (lucascaton) [inline Gemfile]: :tada:

File: app/views/layouts/admin.html.haml [hunk 8]
```diff
@@ -19,4 +20,4 @@
   .container
     = yield
 
-  = render '/shared/footer'
+  = render 'layouts/footer'
```
Comment (lucascaton) [inline app/views/layouts/admin.html.haml]: 👍

File: Gemfile [hunk 9]
```diff
@@ -2,15 +2,13 @@ source 'https://rubygems.org'
...
 
 # Core
-gem 'rails', '>=5.0.0'
+gem 'rails', '5.0.0.1'
 gem 'pg'
 gem 'pg_search'
+gem 'devise'
 gem 'cancancan'
 gem 'paperclip'
 gem 'settingslogic'
```
Comment (lucascaton) [inline Gemfile]: I've been using [`dotenv-rails`](https://github.com/bkeepers/dotenv) exclusively lately, it's so good and works amazingly well with Heroku!

File: .travis.yml [hunk 10]
```diff
@@ -6,6 +6,7 @@ before_install:
 install: bundle install
 before_script:
   - RAILS_ENV=test bundle exec rails db:drop db:create db:schema:load
+bundler_args: --without development
```
Comment (lucascaton) [inline .travis.yml]: without `production` and `staging` too?

Comment (lucascaton) [inline .travis.yml]: `bundler_args: --without development,production,staging`

Comment (ecoologic) [inline .travis.yml]: thanks @lucascaton, I missed that.

File: Gemfile (hunk 9 above)
Comment (lucascaton) [inline Gemfile]: It's not an addition, it'd be a replacement 😄
Get rid of `settingslogic` and add `dotenv-rails`

**Comment by ecoologic** [comment]
@lucascaton I have applied all your requests and also I have checked that `massa` runs smoothly

File: .rubocop.yml [hunk 11]
```diff
@@ -17,6 +17,7 @@ AllCops:
     - 'node_modules/**/*'
     - 'config/**/*'
```
Comment (lucascaton) [inline .rubocop.yml]: Why ignore everything in `config` folder? It'll ignore all initializers!

File: Gemfile (hunk 9 above)
Comment (ecoologic) [inline Gemfile]: we are already using `dot-env`

File: .rubocop.yml (hunk 11 above)
Comment (ecoologic) [inline .rubocop.yml]: Good point, more ideas for your next PR (;

**Comment by ecoologic** [comment]
@lucascaton are you happy with this pr after the changes? - thanks

File: Gemfile (hunk 9 above)
Comment (lucascaton) [inline Gemfile]: Meaning we can remove `settingslogic`? 😄

File: Gemfile [hunk 12]
```diff
@@ -34,25 +32,31 @@ gem 'sass-rails', '~> 5.0'
...
-  # Code analyzers
+  # Code Analyzers
+  gem 'massa'
   gem 'brakeman',             require: false
   gem 'i18n-tasks',           require: false
   gem 'rails_best_practices', require: false
   gem 'rubocop',              require: false
   gem 'rubocop-rspec',        require: false
   gem 'haml_lint',            require: false
+
```
Comment (lucascaton) [inline Gemfile]: Why the line break here?

File: Gemfile [hunk 13]
```diff
@@ -34,25 +32,31 @@ gem 'sass-rails', '~> 5.0'
 # gem "linkedin"
 
 group :development, :test do
-  gem 'pry-byebug'
-  gem 'pry-doc'
-  gem 'pry-rails'
-  gem 'dotenv-rails'
-  gem 'massa'
   gem 'puma'
+  gem 'dotenv-rails'
```
Comment (lucascaton) [inline Gemfile]: It should be available to any envs, including staging and prod

File: Gemfile (hunk 9 above)
Comment (ecoologic) [inline Gemfile]: @lucascaton  IMHO Yes, I don't think it's an essential gem.

File: Gemfile (hunk 13 above)
Comment (ecoologic) [inline Gemfile]: Let's assume we have Heroku, how do you create the `.dotenv.production` file without storing pwds in the repo?

File: Gemfile (hunk 12 above)
Comment (ecoologic) [inline Gemfile]: The top comment (`Code Analyzers`) ends its relevance here.

Comment (lucascaton) [inline Gemfile]: Why not to put the new gem together with the other (before `Code Analyzers` section)? 😄

File: Gemfile (hunk 13 above)
Comment (lucascaton) [inline Gemfile]: You can keep it in 1Password (or any other similar tool).

File: .rubocop.yml [hunk 14]
```diff
@@ -25,34 +26,30 @@ Lint/HandleExceptions:
...
+  Max: 25
 
 Metrics/ClassLength:
   Max: 100
 
 Metrics/LineLength:
   Max: 100
 
+Style/StringLiterals:
+  EnforcedStyle: single_quotes
```
Comment (ecoologic) [inline .rubocop.yml]: 👎 @lucascaton this goes against our own style guide, please, let's not start another unproductive single / double quote war. ):

Comment (lucascaton) [inline .rubocop.yml]: @ecoologic this is your PR man!!! LOL

Comment (buitranquyet) [inline .rubocop.yml]: @lucascaton/@ecoologic I think I added this one when upgraded to Rails 5, Erick just moved it around.

Although both `single_quotes` or `double_quotes` are [both considered as good](https://github.com/net-engine/ruby-style-guide#user-content-consistent-string-literals). I think to make the style to be consistent in the whole project later, we should pick one and `EnforcedStyle`. Trigger/SCOUT enforced single quotes, Outfit enforced double_quotes...

**Pull request merged**